### PR TITLE
Assignment lesson 1.5 is done

### PR DESCRIPTION
**Pull request by omer-jan**
My code is ready for your kind review.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+  return [todoList,setTodoList];
+}
 function App() { 
-
-  const [todoList, setTodoList] = useState([]);
-
+  const[todoList,setTodoList]=useSemiPersistentState()
   let addTodo =(newTodo)=>{ 
-  //  console.log(newTodo);
     setTodoList([...todoList,newTodo])
```
Comment (kwilson05) [inline src/App.js]: So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (setTodoList) and function param will get the most recent value of the state.

So it'd look something like this:

```suggestion
    
     setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

Comment (kwilson05) [reply src/App.js]: Exercise 1.4 overview section actually explains my comment a bit better 😆
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,22 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
-
+const useSemiPersistentState =()=>{
+  // initialTodoList will get the stored list from localStorage if list dose not exist then it will be initialized with empty array 
+  const initialTodoList=JSON.parse(localStorage.getItem('savedTodoList'))==null?[]:JSON.parse(localStorage.getItem('savedTodoList'));
```
Comment (kwilson05) [inline src/App.js:6]: Nice job check for null here 💯 
So javascript is a funky language. You can check for truthy value and assign a value to a variable at the same time
For instance, you could write this line of code like this:
```suggestion
  const initialTodoList=JSON.parse(localStorage.getItem('savedTodoList'))==null?[]:JSON.parse(localStorage.getItem('savedTodoList'));
  
  const initialTodoList =  JSON.parse(localStorage.getItem('savedTodoList')) || []
```
Very funky lol So if the localStorage actual has a value `initialTodoList` will equal the saved/persisted array. If it localStorage returns null. Null == false so the `[]` will be assigned to `initialTodoList`

Gotta love JS lol

Comment (kwilson05) [reply src/App.js:6]: On another note 🎖️ for the `const` :) Nice man!!! Use `const` as much as possible

**Comment by kwilson05** [comment]
Works like a charm!! 🍀 Good job man!! Keep up the good work.

I left some feedback you can address during exercise 1.6 💪

File: src/App.js (hunk 1 above)
Comment (omer-jan) [reply src/App.js]: thank you got it.

**Pull request merged**